src/Microsoft.ML/Runtime/EntryPoints/CrossValidationMacro.cs

```diff
@@ -214,45 +251,141 @@ public sealed class Output
+            overall = EvaluateUtils.CombineFoldMetricsDataViews(env, overall, input.OverallMetrics.Length);
+
+            IDataView conf = null;
+            if (Utils.Size(input.ConfusionMatrix) > 0)
+            {
+                EvaluateUtils.ReconcileSlotNames<double>(env, input.ConfusionMatrix, MetricKinds.ColumnNames.Count, NumberType.R8);
+
+                for (int i = 0; i < input.ConfusionMatrix.Length; i++)
+                {
+                    var idv = input.ConfusionMatrix[i];
+                    // Find the old Count column and drop it.
+                    for (int col = 0; col < idv.Schema.ColumnCount; col++)
+                    {
+                        if (idv.Schema.IsHidden(col) &&
+                            idv.Schema.GetColumnName(col).Equals(MetricKinds.ColumnNames.Count))
+                        {
+                            input.ConfusionMatrix[i] = new ChooseColumnsByIndexTransform(env,
+                                new ChooseColumnsByIndexTransform.Arguments() { Drop = true, Index = new[] { col } }, idv);
+                            break;
+                        }
+                    }
+                }
+                conf = EvaluateUtils.CombineOverallMetrics(env, input.ConfusionMatrix);
+            }
+
+            return new CombinedOutput() { PerInstanceMetrics = perInst[0], OverallMetrics = overall, ConfusionMatrix = conf };
+        }
+
+        private static IMamlEvaluator GetEvaluator(IHostEnvironment env, MacroUtils.TrainerKinds kind)
+        {
+            switch (kind)
+            {
+                case MacroUtils.TrainerKinds.SignatureBinaryClassifierTrainer:
+                    return new BinaryClassifierMamlEvaluator(env, new BinaryClassifierMamlEvaluator.Arguments());
+                case MacroUtils.TrainerKinds.SignatureMultiClassClassifierTrainer:
+                    return new MultiClassMamlEvaluator(env, new MultiClassMamlEvaluator.Arguments());
+                case MacroUtils.TrainerKinds.SignatureRegressorTrainer:
+                    return new RegressionMamlEvaluator(env, new RegressionMamlEvaluator.Arguments());
+                case MacroUtils.TrainerKinds.SignatureRankerTrainer:
+                    return new RankerMamlEvaluator(env, new RankerMamlEvaluator.Arguments());
+                case MacroUtils.TrainerKinds.SignatureAnomalyDetectorTrainer:
+                    return new AnomalyDetectionMamlEvaluator(env, new AnomalyDetectionMamlEvaluator.Arguments());
+                case MacroUtils.TrainerKinds.SignatureClusteringTrainer:
+                    return new ClusteringMamlEvaluator(env, new ClusteringMamlEvaluator.Arguments());
+                case MacroUtils.TrainerKinds.SignatureMultiOutputRegressorTrainer:
+                    return new MultiOutputRegressionMamlEvaluator(env, new MultiOutputRegressionMamlEvaluator.Arguments());
+                default:
+                    throw env.Except($"Trainer kind {kind} does not have an evaluator");
```

Comment:
>Except [](start = 30, length = 6)

This really seems like it wants to be an `ExceptParam` on `kind`. Isn't it great that you have me here to provide these great suggestions?